src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -0,0 +1,421 @@
+        # Verifies whether the line under consideration (i.e. package_details_in_output) contains relevant package details.
+        # package_details_in_output will be of the following format if it is valid
+        #   Sample package details in TDNF:
+        #   python3-libs                       x86_64                    3.12.3-5.azl3                      azurelinux-official-base   36.05M                      10.52M
+        return len(package_details_in_output) == 6 and self.is_arch_in_package_details(package_details_in_output[1], package_arch_to_look_for)
+
+    @staticmethod
+    def is_arch_in_package_details(package_detail, package_arch_to_look_for):
+        # Using a list comprehension to determine if chunk is a package
+        return len([p for p in package_arch_to_look_for if p in package_detail]) == 1
+
+    def get_dependent_list(self, packages):
+        """Returns dependent List for the list of packages"""
+        package_names = ""
+        for index, package in enumerate(packages):
+            if index != 0:
+                package_names += ' '
+            package_names += package
+
+        self.composite_logger.log_verbose("[TDNF] Resolving dependencies. [Command={0}]".format(str(self.single_package_upgrade_simulation_cmd + package_names)))
+        output = self.invoke_package_manager(self.single_package_upgrade_simulation_cmd + package_names)
+        dependencies = self.extract_dependencies(output, packages)
+        self.composite_logger.log_verbose("[TDNF] Resolved dependencies. [Packages={0}][DependencyCount={1}]".format(str(packages), len(dependencies)))
+        return dependencies
+
+    def get_product_name(self, package_name):
+        """Retrieve package name """
+        return package_name
+
+    def get_product_name_and_arch(self, package_name):
+        """Splits out product name and architecture - if this is changed, modify in PackageFilter also"""
+        architectures = Constants.SUPPORTED_PACKAGE_ARCH
+        for arch in architectures:
+            if package_name.endswith(arch):
+                return package_name[:-len(arch)], arch
+        return package_name, None
+
+    def get_product_name_without_arch(self, package_name):
+        """Retrieve product name only"""
+        product_name, arch = self.get_product_name_and_arch(package_name)
+        return product_name
+
+    def get_product_arch(self, package_name):
+        """Retrieve product architecture only"""
+        product_name, arch = self.get_product_name_and_arch(package_name)
+        return arch
+
+    def get_product_name_with_arch(self, package_detail, package_arch_to_look_for):
+        """Retrieve product name with arch separated by '.'. Note: This format is default in tdnf. Refer samples noted within func extract_dependencies() for more clarity"""
+        return package_detail[0] + "." + package_detail[1] if package_detail[1] in package_arch_to_look_for else package_detail[1]
```

Comment:
why package_detail[1] is returned, shouldn't it be package_detail[0] which contains product name.

Reply:
This function returns packagename + arch in the format: <packagename>.<arch>. 
package_detail[1] contains arch